openmdao/core/driver.py

```diff
@@ -523,6 +540,52 @@ def get_constraint_metadata(self):
         """
         return self._cons
 
+    def _gather_vars(self, vec, varnames):
+        '''
+        Gathers and returns only variables listed in 
+        `varnames` from the vector `vec`
+        '''
+        local_vars = []
+
+        for key, value in iteritems(vec):
```

Comment:
Figured it out. `__getitem__` only returns the unflattened value, not all the metadata.